components/calendar/src/iso.rs

```diff
@@ -458,18 +471,38 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1237-L1258
     pub(crate) fn iso_from_fixed(date: i32) -> Date<Iso> {
         let year = Self::iso_year_from_fixed(date);
-        let prior_days = date - Self::iso_new_year(year);
+        // Gets the next possible ISO year
+        let next_year = year.saturating_add(1);
+        // Calculates the fixed days difference between the next year and the current year
+        let year_length = Self::fixed_from_iso_integers(next_year, 12, 31)
+            .unwrap()
+            .saturating_sub(Self::fixed_from_iso_integers(year, 12, 31).unwrap());
+        // Increase the date by the fixed days difference and set to next_year, but only if the current year is MIN_YEAR.
+        let (fixed_date, adjusted_year) = if year == MIN_YEAR {
+            (date.saturating_add(year_length), next_year)
+        } else {
+            (date, year)
+        };
+        // Calculates the prior days of the adjusted year, then applies a correction based on leap year conditions for the correct ISO date conversion.
+        let prior_days = fixed_date.saturating_sub(Self::iso_new_year(adjusted_year));
         #[allow(clippy::unwrap_used)] // valid day and month
-        let correction = if date < Self::fixed_from_iso_integers(year, 3, 1).unwrap() {
+        let correction = if date < Self::fixed_from_iso_integers(adjusted_year, 3, 1).unwrap() {
             0
         } else if Self::is_leap_year(year) {
             1
         } else {
             2
         };
-        let month = quotient(12 * (prior_days + correction) + 373, 367) as u8; // in 1..12 < u8::MAX
+        let month = quotient(
+            (12i32)
+                .saturating_mul(prior_days.saturating_add(correction))
+                .saturating_add(373),
```

Comment:
Nit: You can most likely un-saturate these and put them back to what they were before